server-spi/src/main/java/org/keycloak/models/ClientSessionContext.java

```diff
@@ -43,18 +43,23 @@
 
     /**
      * Returns client scopes as a stream.
-     * @return Stream of client scopes.
+     * @return Stream of client scopes. Never returns {@code null}.
      */
     Stream<ClientScopeModel> getClientScopesStream();
 
     /**
+     * @deprecated Use {@link #getRolesStream() getRolesStream} instead.
      * @return expanded roles (composite roles already applied)
      */
     @Deprecated
     default Set<RoleModel> getRoles() {
         return getRolesStream().collect(Collectors.toSet());
     }
 
+    /**
+     * Returns expanded roles as a stream.
```

Comment:
I found out the meaning of `expanded roles` term only from javadoc above. Maybe it'd be worth to rewrite this javadoc to include the info that it returns all roles included composite ones, wdyt?

Reply:
Done